Automattic/jetpack
# Changelogger: new 'add-pr-num' option

#### Changes proposed in this Pull Request:

Adds a new `add-pr-num` option to the changelogger `write` command. This option tries to grab the PR number from the commit subject message for a changelog file - which should correspond to the PR that introduced that change. The PR number is then appended to each change entry in the written changelog file.

Of note, this requires that the commit subject ends like `(#123)` which GitHub adds by default to the end of a commit subject when using squash merging for pull requests.

This command will help whoever is writing/editing the changelog lookup PRs much easier. Sometimes the changelog entries can be unfamiliar or vague, and trying to search for the corresponding PR otherwise can be tedious.

#### Jetpack product discussion

Internal: p1650404926832289-slack-C01U2KGS2PQ

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* After patching, try running `jetpack changelogger write plugins/jetpack --add-pr-num` and check the written changelog.md file. Each change entry should hopefully end with the PR number that introduced that change like `[#123]`.
* Run without the new option `jetpack changelogger write plugins/jetpack` to make sure PR numbers are _not_ appended and nothing else breaks.
